internal/xds/xdsdepmgr/xds_dependency_manager.go

```diff
@@ -130,15 +170,203 @@ func (m *DependencyManager) annotateErrorWithNodeID(err error) error {
 	return fmt.Errorf("[xDS node id: %v]: %v", m.nodeID, err)
 }
 
-// maybeSendUpdateLocked checks that all the resources have been received and sends
-// the current aggregated xDS configuration to the watcher if all the updates
-// are available.
+// maybeSendUpdateLocked checks that all the resources have been received and
+// sends the current aggregated xDS configuration to the watcher if all the
+// updates are available.
 func (m *DependencyManager) maybeSendUpdateLocked() {
-	m.watcher.Update(&xdsresource.XDSConfig{
+	if m.currentListenerUpdate == nil || m.currentRouteConfig == nil {
+		return
+	}
+	config := &xdsresource.XDSConfig{
 		Listener:    m.currentListenerUpdate,
 		RouteConfig: m.currentRouteConfig,
 		VirtualHost: m.currentVirtualHost,
-	})
+		Clusters:    make(map[string]*xdsresource.ClusterResult),
+	}
+	if !EnableClusterAndEndpointsWatch {
```

Comment:
Nit: Please add a newline above this as this is a logical separate block of code.